protoc-gen-connect-kotlin/src/main/kotlin/build/buf/protocgen/connect/Generator.kt

```diff
@@ -106,28 +101,28 @@ class Generator : CodeGenerator {
 
     private fun serviceClientInterface(
         packageName: String,
-        service: DescriptorProtos.ServiceDescriptorProto
+        service: Descriptors.ServiceDescriptor
     ): TypeSpec {
         val interfaceBuilder = TypeSpec.interfaceBuilder(serviceClientInterfaceClassName(packageName, service))
         val functionSpecs = interfaceMethods(
-            service.methodList
+            service.methods
         )
         return interfaceBuilder
             .addFunctions(functionSpecs)
             .build()
     }
 
     private fun interfaceMethods(
-        methods: List<DescriptorProtos.MethodDescriptorProto>
+        methods: List<Descriptors.MethodDescriptor>
     ): List<FunSpec> {
         val functions = mutableListOf<FunSpec>()
         val headerParameterSpec = ParameterSpec.builder("headers", HEADERS_CLASS_NAME)
             .defaultValue("%L", "emptyMap()")
             .build()
         for (method in methods) {
-            val inputClassName = ClassName.bestGuess(className(method.inputType))
-            val outputClassName = ClassName.bestGuess(className(method.outputType))
-            if (method.hasClientStreaming() && method.hasServerStreaming()) {
+            val inputClassName = ClassName.bestGuess(getClassName(method.inputType))
```

Comment:
I think we'll want to create a helper for this to actually compute the correct class name, instead of using `bestGuess`. Otherwise, if someone is not using expected naming conventions, this could return an incorrect name.

I don't think it's hard either. You basically get the package for the given element's file as the first component. Then you strip that package prefix from the element's full name. And then you split what remains on `.`.  Then you have the package and the sequence of (possibly enclosing) class names that you can supply to `ClassName.get`.

Reply:
Sounds good. I think I was able to pull the right information with this:
```
  private fun classNameFromType(inputType: Descriptors.Descriptor): ClassName {
      if (inputType.file.options.javaMultipleFiles) {
          return ClassName(getFileJavaPackage(inputType.file), inputType.name)
      }
      return ClassName(getFileJavaPackage(inputType.file), getFileClassName(inputType.file), inputType.name)
  }
```